### PR TITLE
use the target as cwd when evaluating git origin

### LINKED ISSUE #4103: --config=auto detects wrong project URL
**Describe the bug**
The --config=auto mode detects the project of the current working directory, not the project in the target repository. 

**Expected behavior**
--config=auto should detect the project URL of the target specified, and possibly enforce that only a single  path target is supplied.

**What is the priority of the bug to you?**

- [ ] P0: blocking your adoption of Semgrep or workflow
- [x] P1: important to fix or quite annoying
- [ ] P2: regular bug that should get fixed

### PR DESCRIPTION
**Pull request by ievans**
Fixes https://github.com/returntocorp/semgrep/issues/4103

Needs careful review by @brendongo though

PR checklist:

- [ ] Documentation is up-to-date
- [ ] Changelog is up-to-date
- [x] Change has no security implications (otherwise, ping security team)

### CODE REVIEW

File: semgrep/semgrep/commands/scan.py [hunk 1]
```diff
@@ -750,7 +750,10 @@ def scan(
                 )
             else:
                 resolved_configs, config_errors = semgrep.config_resolver.get_config(
-                    pattern, lang, config or [], project_url=get_project_url()
+                    pattern,
+                    lang,
+                    config or [],
+                    project_url=get_project_url(target_sequence),
```
Comment (ievans) [inline semgrep/semgrep/commands/scan.py:756]: @brendongo this is the biggest thing I'm not sure about. What should the behavior be here for the scan subcommand?

**Comment by ievans** [comment]
@brendongo pls review?

File: semgrep/semgrep/project.py [hunk 2]
```diff
@@ -12,17 +13,29 @@
...
+        target = targets[0]
+
     try:
         return cast(
             str,
             sub_check_output(
                 ["git", "ls-remote", "--get-url"],
                 encoding="utf-8",
                 stderr=subprocess.DEVNULL,
+                cwd=target,
```
Comment (brendongo) [inline semgrep/semgrep/project.py:38]: I don't think this is right? Does this work if target is not a directory?

Comment (ievans) [reply semgrep/semgrep/project.py:38]: you're right, it throws `NotADirectoryError` if it's a directory

**Pull request closed without merging**